app/javascript/app/components/table.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Table } from 'react-bootstrap';
+
+function THeader(props) {
+  return (
+    <thead>
+      <tr>
+        {props.config.map((field, i) =>
+          <th key={'header-' + field.header}>
+            {field.header}
+          </th>
+        )}
+      </tr>
+    </thead>
+  );
+}
+
+function Row(props) {
+  return (
+    <tr key={'offer-' + props.offer_id + '-row'}>
+      {props.config.map((field, i) =>
+        <td key={'offer-' + props.offer_id + '-row-' + i}>
+          {field.data(props.offer)}
+        </td>
+      )}
+    </tr>
+  );
+}
+
+class AdminTable extends React.Component {
+  render() {
+    return (
+      <Table striped bordered condensed hover>
+        <THeader config={this.props.config} />
+        <tbody>
+          this.props.offers().map((offer) =>
```

Comment:
So, using this Immutable method on a JS object doesn't work.
If instead we try to use JS methods on this object, we get a real issue.

For example, if we change lines 37-38 to
```
{Object.entries(this.props.offers).map(([key, value]) =>
                                   <Row offer_id={key} offer={value} {...this.props} />)}
```
this *would* work on a JS object, but it doesn't work the way that we would expect on an Immutable object.
If we print `Object.entries(this.props.offers)`, we get the following output:
![image](https://user-images.githubusercontent.com/5559014/29373478-64afd0b0-827c-11e7-9e32-83fd412b6b91.png)
You'll notice that JS has treated each of the Immutable object properties as attributes.